pkg/apis/proto/function/reducefn/reduce.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+option go_package = "github.com/numaproj/numaflow-go/pkg/apis/proto/function/reducefn";
```

Comment:
A generic question, any reason of adding a tailing `-fn` to the package?

Reply:
map, reduce package names are already present, so added fn here. I couldn't come up with a better package name :)